### PR TITLE
Introduce beginPlaybackFrom and restartable live facade

### PR DESCRIPTION
**Pull request by rthompson1991**

### CODE REVIEW

File: static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js [hunk 1]
```diff
@@ -346,6 +346,20 @@ window.commonTests.mediaPlayer.cehtml.mixinTests = function (testCase, mediaPlay
         });
     };
 
+    mixins.testBeginPlaybackFromSetsCurrentTimeAndCallsPlayOnMediaElementWhenInStoppedState = function(queue) {
```
Comment (sdclibbery) [inline static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js]: Should there be more tests that assert its an error to call beginPlaybackFrom in the other states?

File: static/script-tests/tests/devices/mediaplayer/live/unseekable.js [hunk 2]
```diff
@@ -84,13 +84,14 @@
     this.LivePlayerSupportLevelUnseekableTest.prototype.testSeekableLivePlayerRemoveAllEventCallbacksCallsFunctionInMediaPlayer = testFunctionsInLivePlayerCallMediaPlayerFunctions('removeAllEventCallbacks', 0);
```
Comment (sdclibbery) [inline static/script-tests/tests/devices/mediaplayer/live/unseekable.js]: Should this have been renamed to 'playable', or is that a separate pull request? :-)

File: static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js (hunk 1 above)
Comment (jmalk) [inline static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js]: @sdclibbery rename to playable is on a branch I've been working on - there may be some tidy up needed to merge both that one and this.

Comment (rthompson1991) [inline static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js]: I was wondering about whether there should be tests around that, but I couldn't see any tests around calling in an invalid state for any other functions.

Comment (rthompson1991) [inline static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js]: Never mind, they're in the common tests. D'oh. I'll add them now.

Comment (rthompson1991) [inline static/script-tests/tests/devices/mediaplayer/cehtmlcommontests.js]: Added some more common tests now

**Pull request merged**